### PR TITLE
AC_Avoid - remove use of inertialnav

### PR DESCRIPTION
**Pull request by peterbarker**
This is an excerpt from #7345 

I expect these commits to be squashed when pushed.

Changes from the original are basically to start to work in metres where possible, to take an AP_AHRS rather than an AP_AHRS_NavEKF and to check the results of the AHRS calls.

### CODE REVIEW

File: libraries/AC_Avoidance/AC_Avoid.cpp [hunk 1]
```diff
@@ -102,8 +102,9 @@ void AC_Avoid::adjust_velocity_z(float kP, float accel_cmss, float& climb_rate_c
     // calculate distance below fence
     if ((_enabled & AC_AVOID_STOP_AT_FENCE) > 0 && (_fence.get_enabled_fences() & AC_FENCE_TYPE_ALT_MAX) > 0) {
         // calculate distance from vehicle to safe altitude
-        float veh_alt = get_alt_above_home();
-        alt_diff_cm = _fence.get_safe_alt_max() * 100.0f - veh_alt;
+        float veh_alt;
+        _ahrs.get_relative_position_D_home(veh_alt);
+        alt_diff_cm = (_fence.get_safe_alt_max() - veh_alt) * 100.0f;
```
Comment (khancyr) [inline libraries/AC_Avoidance/AC_Avoid.cpp]: I think it miss * -1.0f here

Comment (khancyr) [reply libraries/AC_Avoidance/AC_Avoid.cpp]: just for veh_alt

**Comment by khancyr** [comment]
I looks mostly good, I just think it miss a  * -1 when getting _ahrs.get_relative_position_D_home(veh_alt)

**Comment by khancyr** [comment]
And my other consern is that in copter we pass AP_AHRS_NavEKF object on constructor not AP_AHRS

**Comment by peterbarker** [comment]
On Thu, 7 Dec 2017, Pierre Kancir wrote:

> I looks mostly good, I just think it miss a * -1 when getting _ahrs.get_relative_position_D_home(veh_alt)

Please stare at this, and the comment I put in.

We *add* a height limit (up) to a Down position returned by ahrs.  I think 
it is correct :-)  And saves a useless multiplication :-)  If I can get 
away with this with a (better!) comment I'd like to.

**Comment by peterbarker** [comment]
On Thu, 7 Dec 2017, Pierre Kancir wrote:

> And my other consern is that in copter we pass AP_AHRS_NavEKF object on constructor not AP_AHRS

That's not a problem - it simply means that any NavEKF methods are not 
available to AC_Avoid.  But it also means AC_Avoid can be used if you 
don't currently have a NavEKF AHRS object available, or, indeed, are using 
an external AHRS (once we get support for those...)!

Are you OK to fixup all of this stuff in with yours to be a single 
commit?

**Comment by khancyr** [comment]
then https://github.com/ArduPilot/ardupilot/pull/7373/files#diff-ba97cb150b9c1d67a5b4afca880f63e6R107 should be like https://github.com/ArduPilot/ardupilot/pull/7373/files#diff-ba97cb150b9c1d67a5b4afca880f63e6R118 , no ?

**Comment by peterbarker** [comment]
On Thu, 7 Dec 2017, Pierre Kancir wrote:

> then https://github.com/ArduPilot/ardupilot/pull/7373/files#diff-ba97cb150b9c1d67a5b4afca880f63e6R107 should be like
> https://github.com/ArduPilot/ardupilot/pull/7373/files#diff-ba97cb150b9c1d67a5b4afca880f63e6R118 , no ?

Yes, indeed!  Good catch.

Fix pushed in.

**Comment by amilcarlucas** [comment]
Does this allow me to add AC_Avoid to the rover build ???

**Comment by khancyr** [comment]
@amilcarlucas no, it need to remove inertialnav dependency on fence and proximity too

**Comment by rmackay9** [comment]
Ah, I'd forgotten about Fence.  Proximity apparently doesn't use inertial-nav (Peter tells me).

**Comment by peterbarker** [comment]
On Thu, 7 Dec 2017, Francisco Ferreira wrote:

>  *  I'd prefer we don't remove methods from AP_InertialNav one at a time
>     but instead remove the entire library at once. I understand this means
>     leaving an unused method for now, but I hope the other PR will be
>     merged soon enough

I won't remove any more, then :-)

>  *  I don't understand the need to change from AP_AHRS_NavEKF to AP_AHRS,
>     all vehicles use the NavEKF version and this library specifically needs
>     methods from it (which instead are being added to the generic AP_AHRS).

Because if a library can take a more generic version of an object it 
should - and having a height limit placed on the control loops seems a 
reasonable thing for an AHRS to specify.  At some stage we may have a 
pass-through AHRS - and it would be nice if we didn't have to make it a 
NavEKF version to have it all work!

Also, unless we remove AP_AHRS_NAVEKF_AVAILABLE we can't assume we have 
it on all vehicles...

>  *  I'm always requesting squashes, but, besides the first two commits, I
>     wouldn't expect anything else to be squashed

I'm really not fussed either way!

**Pull request merged**